tsl/test/t/001_simple_multinode.pl

```diff
@@ -5,40 +5,22 @@
 # test a simple multi node cluster creation and basic operations
 use strict;
 use warnings;
-use TimescaleNode qw(get_new_ts_node);
+use AccessNode;
+use DataNode;
 use TestLib;
 use Test::More tests => 9;
 
 #Initialize all the multi-node instances
-my @nodes = ();
-foreach my $nodename ('an', 'dn1', 'dn2')
-{
-	my $node = get_new_ts_node($nodename);
-	$node->init;
-	$node->start;
-	# set up the access node
-	if ($node->name eq 'an')
-	{
-		$node->safe_psql('postgres', "CREATE DATABASE an");
-		$node->safe_psql('an',       "CREATE EXTENSION timescaledb");
-	}
-	push @nodes, $node;
-}
+my $an = AccessNode->create('an');
+my $dn1 = DataNode->create('dn1');
+my $dn2 = DataNode->create('dn2');
 
-#Add the data nodes from the access node
-foreach my $i (1 .. 2)
-{
-	my $host = $nodes[$i]->host();
-	my $port = $nodes[$i]->port();
-
-	$nodes[0]->safe_psql('an',
-		"SELECT add_data_node('dn$i', database => 'dn$i', host => '$host', port => $port)"
-	);
-}
+$an->add_data_node($dn1);
+$an->add_data_node($dn2);
```

Comment:
One option here is to use `add_data_nodes` (note plural). The drawback is that it makes it more difficult to provide extra parameters apart from the data node, but most necessary parameters can be extracted from the data node so I am not sure this is necessary.
```suggestion
$an->add_data_nodes($dn1, $dn2);
```

Reply:
IMHO, this current one is enough 👍